proto/internal.go

```diff
@@ -87,3 +91,190 @@ func InternalTimeSeriesDataFromValue(value *Value) (*InternalTimeSeriesData, err
 	}
 	return &ts, nil
 }
+
+// AccumulateFrom accumulates values from the supplied InternalTimeSeriesSample
+// into the existing values of the destination InternalTimeSeriesSample.
+func (dest *InternalTimeSeriesSample) AccumulateFrom(src *InternalTimeSeriesSample) {
+	// Accumulate int values
+	dest.IntCount += src.IntCount
+	if dest.IntCount > 1 {
+		// Keep explicit Max and Min value. This is complicated because these
+		// can be nil in the original messages, in which case the Sum is both the
+		// min and the max.
+		valOrDefault := func(maybeVal *int64, defaultVal int64) int64 {
+			if maybeVal != nil {
+				return *maybeVal
+			}
+			return defaultVal
+		}
+		var (
+			destMax = valOrDefault(dest.IntMax, dest.GetIntSum())
```

Comment:
Looking at this, I'm wondering whether you think it might be overkill to support both Ints and Floats. I know we discussed this before but I don't recall the convincing argument that we support both. I'd be tempted to just support Float64. I'm worried about all of these allocations. hmm.